sdk/playwrighttesting/Azure.Developer.MicrosoftPlaywrightTesting.TestLogger/src/Implementation/BlobService.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Azure.Developer.MicrosoftPlaywrightTesting.TestLogger.Interface;
+using Azure.Storage.Blobs;
+using System.Threading.Tasks;
+
+namespace Azure.Developer.MicrosoftPlaywrightTesting.TestLogger.Implementation
+{
+    internal class BlobService:IBlobService
+    {
+        private readonly ILogger _logger;
+
+        public BlobService(ILogger? logger)
+        {
+            _logger = logger ?? new Logger();
+        }
+
+        public async Task UploadBufferAsync(string uri, string buffer, string fileRelativePath)
+        {
+            try
+            {
+                string cloudFilePath = GetCloudFilePath(uri, fileRelativePath);
+                BlobClient blobClient = new(new Uri(cloudFilePath));
+                byte[] bufferBytes = Encoding.UTF8.GetBytes(buffer);
+                await blobClient.UploadAsync(new BinaryData(bufferBytes), overwrite: true).ConfigureAwait(false);
+                _logger.Info($"Uploaded buffer to {fileRelativePath}");
+            }
+            catch (Exception ex)
+            {
+                _logger.Error($"Failed to upload buffer: {ex}");
+            }
+        }
+        public void UploadBlobFile(string uri, string fileRelativePath, string filePath)
+        {
+            string cloudFilePath = GetCloudFilePath(uri, fileRelativePath);
+            BlobClient blobClient = new(new Uri(cloudFilePath));
```

Comment:
client can come from DI

Reply:
@Sid200026

Reply:
We'll take blob client tests and refactoring in another PR, since straightforward DI injection isn't possible since blob client needs to be initialized for every UploadBlobFile method call instead of once at the constructor level (due to the uri param it takes).